lib/mirage.ml

```diff
@@ -1466,6 +1466,31 @@ module Substitutions = struct
     ] @ blocks @ networks
 end
 
+let opamify s =
+  let b = Buffer.create (String.length s) in
+  String.iter (function
+      | 'a'..'z' | 'A'..'Z' | '0'..'9'
+      | '_' | '-' | '+' as c -> Buffer.add_char b c
+      | '.' -> Buffer.add_char b '-'
+      | _ -> ()
+    ) s;
+  let s' = Buffer.contents b in
+  if String.length s' = 0 then raise (Invalid_argument s);
+  s'
+
+let long_name name target =
+  Fmt.strf "mirage-unikernel-%s-%a" (opamify name) Key.pp_target target
+
+let unikernel_name i target = match Info.output i with
+  | Some o -> opamify (Filename.basename o)
```

Comment:
ic, and its opam file will be `foo.opam`!?  Maybe worth documenting somewhere (but unclear to me where to do this best).